gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/GemFireContainer.java

```diff
@@ -1931,7 +1931,7 @@ final void drop(final GemFireTransaction tran) throws StandardException {
           && (indexManager = (GfxdIndexManager)r.getIndexUpdater()) != null) {
         indexManager.drop(tran);
       }
-      if(r == null){
+      if(this.skipListMap != null){
         releaseIndexMemory();
       }
```

Comment:
I meant that an alternative way to handle could have been moving out the releaseIndexMemory accounting to GfxdIndexManager.drop. Just wanted to confirm that it is not being done there already.